### PR TITLE
[MAINTENANCE] Consistently use type declaration for parameters and return values - part 1

### PR DESCRIPTION
**Pull request by beatrycze-volk**
Change for https://github.com/kitodo/kitodo-presentation/issues/399

### CODE REVIEW

File: Classes/Common/Solr/SearchResult/ResultDocument.php [hunk 1]
```diff
@@ -27,87 +29,89 @@ class ResultDocument
...
      * @param array $highlighting: Array of found highlight elements
      * @param array $fields: Array of fields used for search
      *
      * @return void
      */
-    public function __construct($record, $highlighting, $fields)
+    public function __construct(Document $record, array $highlighting, array $fields)
     {
+        var_dump($record);
+        die();
```
Comment (sebastian-meyer) [inline Classes/Common/Solr/SearchResult/ResultDocument.php]: There is debugging code left. Please remove it! ;o)

File: Classes/Common/Solr/Solr.php [hunk 2]
```diff
@@ -522,15 +522,15 @@ protected function _setParams(array $value)
...
+    public function __get(string $var)
     {
         $method = '_get' . ucfirst($var);
         if (
             !property_exists($this, $var)
             || !method_exists($this, $method)
         ) {
             $this->logger->warning('There is no getter function for property "' . $var . '"');
-            return;
+            return null;
```
Comment (sebastian-meyer) [inline Classes/Common/Solr/Solr.php]: What's the difference between `return;` and `return null;`?

File: Classes/Common/Helper.php [hunk 3]
```diff
@@ -87,7 +87,7 @@ class Helper
      *
      * @return FlashMessageQueue The queue the message was added to
      */
-    public static function addMessage($message, $title, $severity, $session = false, $queue = 'kitodo.default.flashMessages')
+    public static function addMessage(string $message, string $title, int $severity, bool $session = false, string $queue = 'kitodo.default.flashMessages') : FlashMessageQueue
```
Comment (sebastian-meyer) [inline Classes/Common/Helper.php]:
```suggestion
    public static function addMessage(string $message, string $title, int $severity, bool $session = false, string $queue = 'kitodo.default.flashMessages'): FlashMessageQueue
```

File: Classes/Common/MetsDocument.php [hunk 4]
```diff
@@ -81,61 +81,61 @@ final class MetsDocument extends AbstractDocument
...
-    protected $mets;
+    protected \SimpleXMLElement $mets;
 
     /**
      * @access protected
-     * @var string|null URL of the parent document (determined via mptr element), or empty string if none is available
+     * @var string URL of the parent document (determined via mptr element), or empty string if none is available
      */
-    protected $parentHref;
+    protected string $parentHref = "";
```
Comment (sebastian-meyer) [inline Classes/Common/MetsDocument.php]:
```suggestion
    protected string $parentHref = '';
```

File: Classes/Common/Solr/SolrSearch.php [hunk 5]
```diff
@@ -614,13 +615,16 @@ public function submit($start, $rows, $processResults = true)
     /**
      * Find all listed metadata using specified query params.
      *
+<<<<<<< HEAD
```
Comment (sebastian-meyer) [inline Classes/Common/Solr/SolrSearch.php]:
```suggestion
```

File: Classes/Common/Solr/SolrSearch.php [hunk 6]
```diff
@@ -614,13 +615,16 @@ public function submit($start, $rows, $processResults = true)
...
      * @access protected
      *
+=======
+>>>>>>> ddf84fd2 (Consistently use type declaration for parameters and return values)
```
Comment (sebastian-meyer) [inline Classes/Common/Solr/SolrSearch.php]:
```suggestion
```

File: Classes/Controller/AbstractController.php [hunk 7]
```diff
@@ -293,7 +293,7 @@ protected function sanitizeRequestData()
      *
      * @return void
      */
-    protected function setPage() {
+    protected function setPage(): void {
```
Comment (sebastian-meyer) [inline Classes/Controller/AbstractController.php]:
```suggestion
    protected function setPage(): void
    {
```

File: Classes/Controller/AbstractController.php [hunk 8]
```diff
@@ -310,7 +310,7 @@ protected function setPage() {
      *
      * @return void
      */
-    protected function setDefaultPage() {
+    protected function setDefaultPage(): void {
```
Comment (sebastian-meyer) [inline Classes/Controller/AbstractController.php]:
```suggestion
    protected function setDefaultPage(): void
    {
```

File: Classes/Controller/NavigationController.php [hunk 9]
```diff
@@ -33,7 +33,7 @@ class NavigationController extends AbstractController
      *
      * @return void
      */
-    public function pageSelectAction(PageSelectForm $pageSelectForm = NULL) {
+    public function pageSelectAction(PageSelectForm $pageSelectForm = NULL): void {
```
Comment (sebastian-meyer) [inline Classes/Controller/NavigationController.php]:
```suggestion
    public function pageSelectAction(PageSelectForm $pageSelectForm = null): void
    {
```

File: Classes/Eid/PageViewProxy.php [hunk 10]
```diff
@@ -100,7 +100,7 @@ protected function copyHeaders(
         ResponseInterface $fromResponse,
         ResponseInterface $toResponse,
         array $headerNames
-    ) {
+    ): ResponseInterface {
```
Comment (sebastian-meyer) [inline Classes/Eid/PageViewProxy.php:103]:
```suggestion
    ): ResponseInterface
    {
```

File: Classes/Hooks/DataHandler.php [hunk 11]
```diff
@@ -343,7 +343,7 @@ public function processCmdmap_postProcess($command, $table, $id)
                         $doc = AbstractDocument::getInstance($document->getLocation(), ['storagePid' => $document->getPid()], true);
                         if ($document !== null && $doc !== null) {
                             $document->setCurrentDocument($doc);
-                            Indexer::add($document);
+                            Indexer::add($document, $this->getDocumentRepository());
```
Comment (sebastian-meyer) [inline Classes/Hooks/DataHandler.php:346]: Is this change intended or some kind of merging error?

File: Classes/Controller/PageViewController.php [hunk 12]
```diff
@@ -111,7 +111,7 @@ protected function getFulltext($page)
             $physicalStructureInfo = $this->document->getCurrentDocument()->physicalStructureInfo[$this->document->getCurrentDocument()->physicalStructure[$page]];
             $fileId = $physicalStructureInfo['files'][$fileGrpFulltext];
             if (!empty($fileId)) {
-                $file = $this->document->getCurrentDocument()->getFileLocation($fileId);
+                $file = $this->document->getCurrentDocument()->getFileInfo($fileId);
```
Comment (sebastian-meyer) [inline Classes/Controller/PageViewController.php]: Is this change intended or some kind of merging error?

File: Classes/Controller/OaiPmhController.php [hunk 13]
```diff
@@ -714,7 +715,7 @@ protected function generateOutputForDocumentList(array $documentListSet)
         }
         $verb = $this->parameters['verb'];
 
-        $documents = $this->documentRepository->getOaiDocumentList($this->settings, $documentsToProcess);
+        $documents = $this->documentRepository->getOaiDocumentList($documentsToProcess);
```
Comment (sebastian-meyer) [inline Classes/Controller/OaiPmhController.php]: Is this change intended or some kind of merging error?

File: Classes/Updates/FileLocationUpdater.php [hunk 14]
```diff
@@ -19,6 +19,7 @@
 use TYPO3\CMS\Core\Core\Environment;
 use TYPO3\CMS\Core\Database\ConnectionPool;
 use TYPO3\CMS\Core\Database\Query\Expression\ExpressionBuilder;
+use TYPO3\CMS\Core\Log\Logger;
```
Comment (sebastian-meyer) [inline Classes/Updates/FileLocationUpdater.php]: You added this namespace, but removed the `$logger` property, so this is never used. This seems odd.
```suggestion
```

File: Classes/ViewHelpers/JsFooterViewHelper.php [hunk 15]
```diff
@@ -50,7 +51,7 @@ public static function renderStatic(
         array $arguments,
         \Closure $renderChildrenClosure,
         RenderingContextInterface $renderingContext
-    ) {
+    ): void {
```
Comment (sebastian-meyer) [inline Classes/ViewHelpers/JsFooterViewHelper.php]:
```suggestion
    ): void
    {
```

File: Classes/ViewHelpers/MetadataWrapVariableViewHelper.php [hunk 16]
```diff
@@ -66,7 +66,7 @@ public static function renderStatic(
         array $arguments,
         \Closure $renderChildrenClosure,
         RenderingContextInterface $renderingContext
-    ) {
+    ): void {
```
Comment (sebastian-meyer) [inline Classes/ViewHelpers/MetadataWrapVariableViewHelper.php]:
```suggestion
    ): void
    {
```

**Review by sebastian-meyer — CHANGES_REQUESTED** [review]
There are some issues you please need to fix!

In general: You never declared the return values if they are mixed. As far as I know, it's possible to declare it as `mixed` like that:
`public function doSomething(string $parameter): mixed`

Of course it's debatable if declaring a return value as `mixed` is useful at all, but I think it's helpful for consistency. What do you think?

File: Classes/Controller/PageViewController.php (hunk 12 above)
Comment (beatrycze-volk) [reply Classes/Controller/PageViewController.php]: It is necessary, in some point I have made error while solving merging conflicta which got already to master and now it reverses it to the correct state.

File: Classes/Controller/OaiPmhController.php (hunk 13 above)
Comment (beatrycze-volk) [reply Classes/Controller/OaiPmhController.php]: Not sure how it has landed here but looks like correct thing to do considering that `public function getOaiDocumentList(array $documentsToProcess): DriverResultStatement` doesn't take two parameters.

File: Classes/Common/Solr/Solr.php (hunk 2 above)
Comment (beatrycze-volk) [reply Classes/Common/Solr/Solr.php]: It should not be changed

**Comment by beatrycze-volk** [comment]
> In general: You never declared the return values if they are mixed. As far as I know, it's possible to declare it as `mixed` like that: `public function doSomething(string $parameter): mixed`
> 
> Of course it's debatable if declaring a return value as `mixed` is useful at all, but I think it's helpful for consistency. What do you think?

It is possible to return `mixed` but first since PHP 8.0: https://php.watch/versions/8.0/mixed-type

**Comment by beatrycze-volk** [comment]
The problem with OAI tests is related to some changes inside the Controller and Domain package, so I have taken those two back and will handle them in seprated PR. This one is now ready for review.

File: Classes/Common/Solr/Solr.php [hunk 17]
```diff
@@ -464,9 +464,9 @@ protected function _getReady()
      *
      * @access protected
      *
-     * @return \Solarium\Client Apache Solr service object
+     * @return Client Apache Solr service object
      */
-    protected function _getService()
+    protected function _getService() : Client
```
Comment (sebastian-meyer) [inline Classes/Common/Solr/Solr.php]:
```suggestion
    protected function _getService(): Client
```

**Review by sebastian-meyer — APPROVED** [review]
